wordpress-mobile/WordPress-Android
# Fixes: Site icon loading multiple times with a flickering effect

Fixes #16155

Fixes Flickering effect in Site Header Icon by rendering the site header only if there is any change in the site header data or if the site header is invisible. The site header will be invisible by default. The site header invisible condition for rendering the site header data takes care of the configuration change scenario. 

If there is any change in the configuration change such as changing to landscape or portrait. The fragment gets loaded again with default XML in which the header is invisible, note in this scenario there won't be any change in the site header data. 

After fix 

https://user-images.githubusercontent.com/17463767/161037175-571f4894-3b2c-40a7-b419-05b7c37d4a77.mp4


To test:
- Launch the app.
- Go to My Site -> Me -> App Settings -> Debug settings.
- Turn on the MySiteDashboardTabsFeatureConfig flag under the Features in development section.
- Restart the app
- Notice that the Site Info is shown above the Tabs
- Notice that Site Icon doesn't while changing the tabs
- Notice that changing the Orientation to Portrait or Landscape works as Expected 


## Regression Notes
1. Potential unintended areas of impact
Site header details not being shown properly 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual test, Unit tests


3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
